CommonTools/RecoAlgos/interface/FKDTree.h

```diff
@@ -0,0 +1,386 @@
+#ifndef COMMONTOOLS_RECOALGOS_FKDTREE_H
+#define COMMONTOOLS_RECOALGOS_FKDTREE_H
+
+#include <vector>
+#include <array>
+#include <algorithm>
+#include <cmath>
+#include <utility>
+#include "FKDPoint.h"
+#include "FQueue.h"
+
+
+namespace{
+    const std::array<unsigned int,32> MultiplyDeBruijnBitPosition{{0, 9, 1, 10, 13, 21, 2, 29, 11, 14, 16, 18, 22, 25, 3, 30,
+  8, 12, 20, 28, 15, 17, 24, 7, 19, 27, 23, 6, 26, 5, 4, 31}};
+    unsigned int ilog2 (unsigned int v)
+	    {
+	    v |= v >> 1; // first round down to one less than a power of 2 
+	    v |= v >> 2;
+	    v |= v >> 4;
+	    v |= v >> 8;
+	    v |= v >> 16;
+	    return MultiplyDeBruijnBitPosition[(unsigned int)(v * 0x07C4ACDDU) >> 27];
+
```

Comment:
Please improve readability by removing the many empty lines (when not needed to separate methods or logical parts inside the code)